web/viewer.js

```diff
@@ -211,8 +211,12 @@ function webViewerLoad() {
       // set various `AppOptions`, by dispatching an event once all viewer
       // files are loaded but *before* the viewer initialization has run.
       const event = document.createEvent("CustomEvent");
-      event.initCustomEvent("webviewerloaded", true, true, {});
-      document.dispatchEvent(event);
+      event.initCustomEvent("webviewerloaded", true, true, {
+        source: window,
+      });
+      // Dispatch the event at the top-level `document`, to support cases where
+      // the viewer is embedded in a *dynamically* created <iframe> element.
+      top.document.dispatchEvent(event);
```

Comment:
`top.document` will throw if top is a cross-origin window. Not sure if that's relevant here and/or should be considered.